frontend/src/lib/components/VisibilitySensor/VisibilitySensor.tsx

```diff
@@ -0,0 +1,23 @@
+import { useActions } from 'kea'
+import React, { useEffect, useRef } from 'react'
+import { visibilitySensorLogic } from './visibilitySensorLogic'
+
+interface VisibilityProps {
+    id: string
+    offset: number
+    children: React.ReactNode | null
+}
+
+export function VisibilitySensor({ id, offset, children }: VisibilityProps): JSX.Element {
+    const { scrolling } = useActions(visibilitySensorLogic({ id, offset }))
+
+    const ref = useRef<HTMLDivElement | null>(null)
+
+    useEffect(() => {
```

Comment:
This was kinda tricky to sort out. I can't do most of these things in a logic because I couldn't find a good way of passing a `ref` to a logic. 

Unsure if there's a better way?

Reply:
Instead of ref, can you just pass an action that sets the element you're interested in?

Reply:
I wasn't sure if that would live past re-renders.

Reply:
I'm not sure tbh, it looks like what is done here however: https://github.com/PostHog/posthog/blob/use-taxo-filter/frontend/src/toolbar/elements/elementsLogic.ts#L27:L27

Reply:
Could use @alexkim205 or @mariusandra s opinion here, before I dive down this rabbit hole 😅 .